### PR TITLE
request legend in json format

### PR DESCRIPTION
**Pull request by stijn-at-work**
Fixes https://github.com/mapproxy/mapproxy/issues/540

### CODE REVIEW

**Comment by drnextgis** [comment]
I don't think that "application/json" should be part of the `image_formats` list. I would say that introducing something like "legend_formats" could be better way to tackle the issue.

**Comment by stijn-at-work** [comment]
I have added the functionality of retrieving json. 

**manual test**
I have tested the functionality manually by adding my own layer to yaml file, running the proxy and making a request in postman. Add image/png to format query param still returns an image. Add application/json to format query param returns json.

**automated test**
I tried to write a test in test/system/test_legendgraphic.py, but did not understand the mock_httpd class. I need some help here. Greetings, Stijn

**Comment by stijn-at-work** [comment]
All tests in test_legendgraphic.py pass locally, since my last commit. How do i rerun the checks of the pull request? 

(Sorry if i ask a lot of questions, but this is the first time i 've made a pull request on an open source Github repo.)

File: mapproxy/client/wms.py [hunk 1]
```diff
@@ -199,22 +199,31 @@ def __init__(self, request_template, http_client=None):
 
     def get_legend(self, query):
         resp = self._retrieve(query)
-        format = split_mime_type(query.format)[1]
```
Comment (weskamm) [inline mapproxy/client/wms.py:202]: why did you remove this code? This is a change which may lead to problems because now you you are giving something like `image/png` to the `ImageSource` in line 206, while it was something like `png` before. Did you check if raster legends still work in different formats?

File: mapproxy/client/wms.py [hunk 2]
```diff
@@ -199,22 +199,31 @@ def __init__(self, request_template, http_client=None):
...
-        self._check_resp(resp)
-        return ImageSource(resp, image_opts=ImageOptions(format=format))
+        self._check_resp(resp, query.format)
+        if query.format == 'json':
+            return resp
+        else:
+            return ImageSource(resp, image_opts=ImageOptions(format=query.format))
```
Comment (weskamm) [inline mapproxy/client/wms.py:206]: see comment above, you changed the input to `query.format` which affects other requests as well

**Review by weskamm — CHANGES_REQUESTED** [review]
Please remove the empty file `mapproxy/test/system/pytest.old/__init__.py`

File: mapproxy/client/wms.py (hunk 1 above)
Comment (stijn-at-work) [reply mapproxy/client/wms.py:202]: when i stepped through the code with the debugger I found out that query.format equals 'png' (or jpeg or gif or json) and not 'image/png'. Therefore splitting the mime-type is useless (splitting does not return an error, it simply returns itself).

Comment (stijn-at-work) [reply mapproxy/client/wms.py:202]: i really don't mind resetting the code that splits the mimetype. done

**Comment by stijn-at-work** [comment]
I fixed the code inline with the change request. Could you review the code again? Thanks, Stijn

**Review by weskamm — APPROVED** [review]
Changes looking good now. Thanks for your contributions!
